crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/continued_indentation.rs

```diff
@@ -0,0 +1,797 @@
+///
+/// Use instead:
+/// ```python
+/// print("Python", (
+///     "Rules"))
+/// ```
+///
+/// ## Formatter compatibility
+/// We recommend against using this rule alongside the [formatter]. The
+/// formatter enforces consistent indentation, making the rule redundant.
+///
+/// [PEP 8]: https://peps.python.org/pep-0008/#indentation
+/// [formatter]:https://docs.astral.sh/ruff/formatter/
+#[violation]
+pub struct ContinuationUnderIndentedHanging;
+
+impl Violation for ContinuationUnderIndentedHanging {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Continuation line under-indented for hanging indent".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks that a continuation line is not indented as far as it should be or
+/// is indented too far.
+///
+/// ## Why is this bad?
+/// According to [PEP8], continued lines with hanging indents should be aligned
+/// to the next indentation level.
+///
+/// ## Example
+/// ```python
+/// print("Python", (
+/// "Rules"))
+/// ```
+///
+/// Use instead:
+/// ```python
+/// print("Python", (
+///     "Rules"))
+/// ```
+///
+/// ## Formatter compatibility
+/// We recommend against using this rule alongside the [formatter]. The
+/// formatter enforces consistent indentation, making the rule redundant.
+///
+/// [PEP 8]: https://peps.python.org/pep-0008/#indentation
+/// [formatter]:https://docs.astral.sh/ruff/formatter/
+#[violation]
+pub struct ContinuationOverIndentedOrMissing;
+
+impl Violation for ContinuationOverIndentedOrMissing {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Continuation line missing indentation or outdented".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks that closing brackets match the same indentation level of the line
+/// that their opening bracket started on.
+///
+/// ## Why is this bad?
+/// According to [PEP8], closing brackets on their own line should match the
+/// same indentation level as the opening bracket.
+///
+/// ## Example
+/// ```python
+/// result = function_that_takes_arguments(
+///     'a', 'b', 'c',
+///     'd', 'e', 'f',
+///     )
+/// ```
+///
+/// Use instead:
+/// ```python
+/// result = function_that_takes_arguments(
+///     'a', 'b', 'c',
+///     'd', 'e', 'f',
+/// )
+/// ```
+///
+/// ## Formatter compatibility
+/// We recommend against using this rule alongside the [formatter]. The
+/// formatter enforces consistent indentation, making the rule redundant.
+///
+/// [PEP 8]: https://peps.python.org/pep-0008/#indentation
+/// [formatter]:https://docs.astral.sh/ruff/formatter/
+#[violation]
+pub struct ClosingBracketMismatched;
+
+impl Violation for ClosingBracketMismatched {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Closing bracket does not match indentation of opening bracket's line".to_string()
+    }
+}
+
+/// ## What it does
+/// Closing brackets should match the indentation of the opening bracket.
+///
+/// ## Why is this bad?
+/// TODO
```

Comment:
flagging myself to actually finish off the docs

Reply:
Since I had to actually put some work in to get the rebase to work, I've done that.